File: T2.2 Manejo de Archivos/SistemaNotas/SistemaNotas.java
import java.io.File;
import java.util.Scanner;

public class SistemaNotas {
    public static void main(String[] args) throws Exception {

        File archivo = new File("estudiantes.txt");

        if (!archivo.exists()) {
            System.out.println("Error: archivo no existe");
            return;
        }

        Scanner sc = new Scanner(archivo);

        int total = 0, aprobados = 0;
        double suma = 0;

        while (sc.hasNext()) {
            String nombre = sc.next();
            double n1 = sc.nextDouble();
            double n2 = sc.nextDouble();
            double n3 = sc.nextDouble();

            double promedio = (n1 + n2 + n3) / 3;

            System.out.println("Nombre: " + nombre);
            System.out.println("Promedio: " + promedio);

            //NUEVA CLASIFICACIÓN
            if (promedio >= 9) {
                System.out.println("Estado: Excelente");
                aprobados++;
            } 
            else if (promedio >= 7) {
                System.out.println("Estado: Bueno");
                aprobados++;
            } 
            else {
                System.out.println("Estado: Reprobado");
            }

            System.out.println("---------------------");

            suma += promedio;
            total++;
        }

        //RESULTADOS FINALES
        if (total > 0) {
            System.out.println("Total estudiantes: " + total);
            System.out.println("Total aprobados: " + aprobados);
            System.out.println("Promedio general: " + (suma / total));
        }

        sc.close();
    }
}

File: T2.2 Manejo de Archivos/SistemaNotas/SistemaNotas.cpp
#include <iostream>
#include <fstream>
using namespace std;

int main() {
    ifstream archivo("estudiantes.txt");

    if (!archivo) {
        cout << "Error: archivo no existe\n";
        return 1;
    }

    string nombre;
    double n1, n2, n3, promedio;
    int total = 0, aprobados = 0;
    double suma = 0;

    while (archivo >> nombre >> n1 >> n2 >> n3) {

        promedio = (n1 + n2 + n3) / 3;

        cout << "Nombre: " << nombre << endl;
        cout << "Promedio: " << promedio << endl;

        //NUEVA CLASIFICACIÓN
        if (promedio >= 9) {
            cout << "Estado: Excelente\n";
            aprobados++;
        }
        else if (promedio >= 7) {
            cout << "Estado: Bueno\n";
            aprobados++;
        }
        else {
            cout << "Estado: Reprobado\n";
        }

        cout << "---------------------\n";

        suma += promedio;
        total++;
    }

    //RESULTADOS FINALES
    if (total > 0) {
        cout << "Total estudiantes: " << total << endl;
        cout << "Total aprobados: " << aprobados << endl;
        cout << "Promedio general: " << suma / total << endl;
    }

    archivo.close();
    return 0;
}
